### ISSUE TITLE
Use injected OSD vs OpenSeadragon

### ISSUE THREAD

**Issue by nselikoff** [msalsbery/OpenSeadragonImagingHelper#31]
Thanks for this helper! I'm trying this out in a ES6 module context so I'm changing the IIFE to a named export and calling it with an instance of `OpenSeadragon` that I import from OpenSeadragon's npm module. In doing this I discovered there are a couple spots where a global `OpenSeadragon` is used instead of the injected `OSD` variable from the IIFE. E.g. https://github.com/msalsbery/OpenSeadragonImagingHelper/blob/b37a4f07f1af5f395c6b02fae89f74be51d100dd/src/imaginghelper.js#L139-L140

**Comment by iangilman** [comment]
Good catch! If you want to make a PR with that change (using the injected OSD variable), I would be happy to merge it. 

It's probably best to leave this as an IIFE, so don't include that change in your PR. What would be awesome is if we could provide something that supported both modes, but I'm not aware of a way to do that in the same file; are you?

**Comment by nselikoff** [comment]
Happy to make a PR. 

Re: multiple module types, I'm importing the main OpenSeadragon via npm... haven't looked at how it's tackled there but maybe could mimic that approach?

**Comment by msalsbery** [comment]
@nselikoff  PRs always very welcome!

I’m tied up with personal issues but should be back on this stuff soon, but I did want to at least drop a little info to save you some work 

1) this is a legacy repository and no longer supported. You are welcome to make changes if it helps you but ai highly recommend using the new repositories in the [openseadragon-imaging organization](https://github.com/openseadragon-imaging) (note the repo names on the newer ones gave hyphens). The new repos build with webpack and babel etc to modules that can be used in commonjs es modules or included in html script tags. They can be, and are deployed to and available from npm, and work great with react etc.

The worst news is this particular plugin repo (imaging helper) needs the most work to refresh it for modern OpenSeadragon, mostly multiple image and World coordinates support). I have a good start on it but broke away before I finished it. If it currently suits your needs, I’d love and welcome any help!

Most of all, this repository [openseadragon-imaginghelper](https://github.com/openseadragon-imaging/openseadragon-imaginghelper) is the official repository that WILL be kept up with OpenSeadragon, so it’s a much better one to use  :)

Let me know whenever you need any answers or if there’s some wy I can possibly help. I am watching github notifications and gradually getting back on this long overdue work

**Comment by msalsbery** [comment]
@iangilman Thank you, beyond words…I will be in touch..

**Comment by msalsbery** [comment]
@nselikoff

Looks like I have the code in local branches for all my plugins that I updated for OSD 3.0 and was going to push when that version released. I’ll get the imaging helper code pushed so you can at least start with the most recent! So sorry for the hassle

**Comment by nselikoff** [comment]
Thanks for the additional info, I had no idea the other repos existed! Maybe the links from the OSD plugin page need updating. PS no rush, I was just trying some stuff out and ended up getting the transforms I needed from the main OSD viewport class so might not need the imaging helper for this current project anyway.

**Comment by msalsbery** [comment]
> Thanks for the additional info, I had no idea the other repos existed! Maybe the links from the OSD plugin page need updating. PS no rush, I was just trying some stuff out and ended up getting the transforms I needed from the main OSD viewport class so might not need the imaging helper for this current project anyway.

Total facepalm on the OSD plugin list page...part of what was supposed to happen when OSD 3.0 was released.

Indeed, most if not all of the transform functions are in OSD. The history of this plugin goes way back for me as I was doing medical  imaging from Microsoft DOS (pre-Windows) through Microsoft MFC, WPF, Silverlight, and finally all-in on the web, where I discovered OpenSeadragon (conveniently a relative of Silverlight's tiled image control which I had been using successfully until browser plugins like Silverlight were taken away. 

So short story long, I had a lot of imaging code cariied along for decades across languges C, C++, C#, and finally javascript. The imaging helper plugin was written mostly just for me so I could use the same code without rewriting transformations to another coordinate system.

Glad you found a solution though, and thank you for the heads up on my code neglect!

Cheers
